test/mock-server-tester.ts

```diff
@@ -0,0 +1,126 @@
+// Copyright 2025 Google LLC
```

Comment:
This file contains utilities for working with the mock server that include a function for producing a deterministic set of errors and a function for shutting down the mock server.